Fix enum ConnectionType and use it in examples for testing

bernhardkaindl opened the pull request:
Add a dictionary for getting the ConnectionType when having a DeviceType.
Used for lookup of the available connections for a given device.

Update: Replaced with a fix for a small issue in the enum ConnectionType and a real-world example of it's use.

igo95862 commented:
Hello,

I am pretty sure that `getattr(ConnectionType, name)` provides same functionality with no code duplication.

```python
getattr(sdbus_async.networkmanager.ConnectionType, 'VETH') == 'veth'
# True
```

bernhardkaindl commented:
@igo95862 Thanks! I noticed that a small fix is needed to make this work for Ethernet devices.

Please find the PR now updated to fix and use `getattr(sdbus_async.networkmanager.ConnectionType, DeviceType(id).name) as suggested with the `netdevinfo` example updated to test it's use on real systems.

We added the enum sdbus_async.networkmanager.ConnectionType very recently, so it's not released yet and we can make this fix. I also checked my code and wasn't using ConnectionType.WIRED for anything yet.
